msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+    Spectrum<Peak> siriusMs2;
+    MsSpectrum msdkSpectrumMs2 = ms2.remove(0);
+
+    double mz[] = msdkSpectrumMs2.getMzValues();
+    double intensity[] = LocalArrayUtil.convertToDoubles(msdkSpectrumMs2.getIntensityValues());
+
+    siriusMs2 = sirius.wrapSpectrum(mz, intensity);
+    String ionization = ion.getName();
+    PrecursorIonType precursor = sirius.getPrecursorIonType(ionization);
+
+    /* MutableMs2Experiment allows to specify additional fields and it is exactly what comes from .getMs2Experiment */
+    MutableMs2Experiment experiment = (MutableMs2Experiment) sirius.getMs2Experiment(parentMass, precursor, null, siriusMs2);
+
+    loadSpectra(experiment);
+
+
+    /* Manual setting up annotations, because sirius.identify does not use some of the fields */
+    sirius.setAllowedMassDeviation(experiment, deviation);
+    sirius.enableRecalibration(experiment, true);
+    sirius.setIsotopeMode(experiment, IsotopePatternHandling.both);
+    if (constraints != null)
+      sirius.setFormulaConstraints(experiment, constraints);
+
+    logger.debug("Sirius starts processing MsSpectrums");
+    List<IdentificationResult> siriusResults  = sirius.identify(experiment,
+        numberOfCandidates, true, IsotopePatternHandling.both, constraints);
+    logger.debug("Sirius finished processing and returned {} results", siriusResults.size());
+
+    return siriusResults;
+  }
+
+  /**
+   * <p> Method for transformation of MsSpectrum into Spectrum<Peak></p>
+   * Method transforms MSDK object into Sirius object
+   * @param msdkSpectrum - non-null MsSpectrum object.
+   * @return new Spectrum<Peak>
+   */
+  private Spectrum<Peak> transformSpectrum(@Nonnull MsSpectrum msdkSpectrum) {
+    double mz[], intensity[];
+    mz = msdkSpectrum.getMzValues();
+    intensity = LocalArrayUtil.convertToDoubles(msdkSpectrum.getIntensityValues());
+    return sirius.wrapSpectrum(mz, intensity);
+  }
+
+  /**
+   * <p> Method for loading MS & MS/MS spectra</p>
+   * Method loads MS and MS/MS spectra into Ms2Experiment object
+   * @param experiment - experiment Sirius works with and where to load spectra
+   */
+  private void loadSpectra(MutableMs2Experiment experiment) {
+    List<MutableMs2Spectrum> ms2spectra = experiment.getMs2Spectra();
+    List<SimpleSpectrum> ms1spectra = experiment.getMs1Spectra();
+
+    if (ms1 != null) {
+      for (MsSpectrum msdkSpectrum : ms1) {
+        Spectrum<Peak> peaks = transformSpectrum(msdkSpectrum);
+        ms1spectra.add(new SimpleSpectrum(peaks));
+      }
+    }
+
+    if (ms2.size() > 0) {
+      for (MsSpectrum msdkSpectrum : ms2) {
+        Spectrum<Peak> peaks = transformSpectrum(msdkSpectrum);
+      /* MutableMs2Experiment does not accept Ms1 as a Spectrum<Peak>, so there is a new object */
+        MutableMs2Spectrum siriusMs2 = new MutableMs2Spectrum(peaks);
+        ms2spectra.add(siriusMs2);
+      }
+    }
+  }
+
+  @Nullable
+  @Override
+  public Float getFinishedPercentage() {
+    return null;
```

Comment:
Is there any way to estimate the progress of the whole operation?

Reply:
Sirius module does not have this opportunity.